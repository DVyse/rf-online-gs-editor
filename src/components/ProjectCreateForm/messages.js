/*
 * ProjectCreateForm Messages
 *
 * This contains all the text for the ProjectCreateForm component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  Submit: {
    id: 'app.components.ProjectCreateForm.Submit',
    defaultMessage: 'Submit',
  },
  Title: {
    id: 'app.components.ProjectCreateForm.Title',
    defaultMessage: 'Title',
  },
  Description: {
    id: 'app.components.ProjectCreateForm.Description',
    defaultMessage: 'Description',
  },
  IsPublicProject: {
    id: 'app.components.ProjectCreateForm.IsPublicProject',
    defaultMessage: 'Is public project',
  },
});
