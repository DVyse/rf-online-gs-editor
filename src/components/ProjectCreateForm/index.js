/**
 *
 * ProjectCreateForm
 *
 */

import React from 'react';
import { Icon, Button, Checkbox, Form } from 'semantic-ui-react';
import PropTypes from 'prop-types';
// import styled from 'styled-components';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

function ProjectCreateForm({
  size,
  loading,
  values,
  onSubmit,
  onChangeTitle,
  onChangeDescription,
  onChangeIsPublic,
}) {
  return (
    <Form size={size} loading={loading}>
      <Form.Field>
        <label>
          <FormattedMessage {...messages.Title} />
        </label>
        <input value={values.title} onChange={onChangeTitle} />
      </Form.Field>
      <Form.Field>
        <label>
          <FormattedMessage {...messages.Description} />
        </label>
        <textarea value={values.description} onChange={onChangeDescription} />
      </Form.Field>
      <Form.Field>
        <Checkbox
          value={1}
          checked={values.isPublic}
          label={
            <label>
              <FormattedMessage {...messages.IsPublicProject} />
            </label>
          }
          type="checkbox"
          onChange={onChangeIsPublic}
        />
      </Form.Field>

      <Button type="submit" primary onClick={onSubmit}>
        <Icon name="plus" />
        <FormattedMessage {...messages.Submit} />
      </Button>
    </Form>
  );
}

ProjectCreateForm.propTypes = {
  size: PropTypes.oneOf([
    'mini',
    'tiny',
    'small',
    'large',
    'big',
    'huge',
    'massive',
  ]),
  loading: PropTypes.bool,
  values: PropTypes.shape({
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    isPublic: PropTypes.bool.isRequired,
  }).isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChangeTitle: PropTypes.func.isRequired,
  onChangeDescription: PropTypes.func.isRequired,
  onChangeIsPublic: PropTypes.func.isRequired,
};

ProjectCreateForm.defaultProps = {
  size: 'small',
  loading: false,
};

export default ProjectCreateForm;
