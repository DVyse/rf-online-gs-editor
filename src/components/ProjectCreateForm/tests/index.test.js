import React from 'react';
import { shallow } from 'enzyme';

import ProjectCreateForm from '../index';

describe('<ProjectCreateForm />', () => {
  it('Should exist', () => {
    const wrapper = shallow(
      <ProjectCreateForm
        values={{
          title: 'title',
          description: 'description',
          isPublic: false,
        }}
        onSubmit={() => undefined}
        onChangeTitle={() => undefined}
        onChangeDescription={() => undefined}
        onChangeIsPublic={() => undefined}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
