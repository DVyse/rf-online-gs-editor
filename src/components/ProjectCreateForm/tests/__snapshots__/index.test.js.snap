// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectCreateForm /> Should exist 1`] = `
<Form
  as="form"
  loading={false}
  size="small"
>
  <FormField>
    <label>
      <FormattedMessage
        defaultMessage="Title"
        id="app.components.ProjectCreateForm.Title"
        values={Object {}}
      />
    </label>
    <input
      onChange={[Function]}
      value="title"
    />
  </FormField>
  <FormField>
    <label>
      <FormattedMessage
        defaultMessage="Description"
        id="app.components.ProjectCreateForm.Description"
        values={Object {}}
      />
    </label>
    <textarea
      onChange={[Function]}
      value="description"
    />
  </FormField>
  <FormField>
    <Checkbox
      checked={false}
      label={
        <label>
          <FormattedMessage
            defaultMessage="Is public project"
            id="app.components.ProjectCreateForm.IsPublicProject"
            values={Object {}}
          />
        </label>
      }
      onChange={[Function]}
      type="checkbox"
      value={1}
    />
  </FormField>
  <Button
    as="button"
    onClick={[Function]}
    primary={true}
    role="button"
    type="submit"
  >
    <Icon
      as="i"
      name="plus"
    />
    <FormattedMessage
      defaultMessage="Submit"
      id="app.components.ProjectCreateForm.Submit"
      values={Object {}}
    />
  </Button>
</Form>
`;
